app/controllers/insecure_password_controller.rb

```diff
@@ -0,0 +1,7 @@
+class InsecurePasswordController < ApplicationController
+  before_action :authenticate_user!
+
+  def show
+    redirect_to user_root_path unless current_user.banned_password_match
```

Comment:
Is this necessary? We only arrive here after checking that the password is in the list:
```ruby
# ...
elsif login_state.user.banned_password_match
  insecure_password_interstitial_path
# ...
```

Reply:
It's arguably not necessary, but someone could also go straight to that page (eg by bookmarking it)

Reply:
That's a good point 👍